File: leetcode/1838-frequency-of-the-most-frequent-element/solution.cpp
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int maxFrequency(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int left = 0;
        long long windowSum = 0;
        int best = 1;
        for (int right = 0; right < (int)nums.size(); right++) {
            windowSum += nums[right];
            while ((long long) nums[right] * (right - left + 1) - windowSum > k) {
                windowSum -= nums[left];
                left++;
            }
            best = max(best, right - left + 1);
        }
        return best;
    }
};

File: leetcode/1838-frequency-of-the-most-frequent-element/Solution.java
import java.util.Arrays;

class Solution {
    public int maxFrequency(int[] nums, int k) {
        Arrays.sort(nums);
        int left = 0;
        long windowSum = 0;
        int best = 1;
        for (int right = 0; right < nums.length; right++) {
            windowSum += nums[right];
            while ((long) nums[right] * (right - left + 1) - windowSum > k) {
                windowSum -= nums[left];
                left++;
            }
            best = Math.max(best, right - left + 1);
        }
        return best;
    }
}
